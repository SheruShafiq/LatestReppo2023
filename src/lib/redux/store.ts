import { configureStore } from "@reduxjs/toolkit";
import sessionSlice from "./slices/sessionSlice";
import layoutSlice from "./slices/layoutSlice";
import mutationSlice from "./slices/mutationSlice";


export const store = configureStore({
  reducer: {
    session: sessionSlice,
    layout: layoutSlice,
    mutation: mutationSlice,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
