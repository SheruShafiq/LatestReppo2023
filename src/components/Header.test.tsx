import ButtonAppBar from "../components/Header";
import { fireEvent, render } from "@testing-library/react";
import matchers from "@testing-library/jest-dom";
import { debug } from "vitest-preview";
expect.extend(matchers);
import useResizeHandler from "../lib/hooks/useResizeHandler";
import { Provider } from "react-redux";
import { store } from "../lib/redux/store";

describe("ButtonAppBar", () => {
  it("renders the component correctly", async () => {
    render(
      <Provider store={store}>
        <ButtonAppBar />
      </Provider>
    );

    debug();
    expect(document.getElementById("headerBox")).toBeInTheDocument();
    expect(document.getElementById("headerAppBar")).toBeInTheDocument();
  });

  it("toggles the drawer state when menu button is clicked", async () => {
    const { getByLabelText } = render(
      <Provider store={store}>
        <ButtonAppBar />
      </Provider>
    );

    const menuButton = getByLabelText("menu");
    await fireEvent.click(menuButton);

    // Verify that the drawer state has changed (e.g., by checking if a CSS class or an attribute has been added/removed)
  });

  it("toggles the account menu state when account button is clicked", async () => {
    const { getByLabelText } = render(
      <Provider store={store}>
        <ButtonAppBar />
      </Provider>
    );

    const accountButton = getByLabelText("account");
    await fireEvent.click(accountButton);

    // Verify that the account menu state has changed (e.g., by checking if a CSS class or an attribute has been added/removed)
  });

  it("renders the menu button when the viewport width is less than or equal to 900", async () => {
    // Mock the useResizeHandler hook to return a width of 900 or less
    const { getByLabelText } = render(
      <Provider store={store}>
        <ButtonAppBar />
      </Provider>
    );

    const menuButton = getByLabelText("menu");
    expect(menuButton).toBeInTheDocument();
    expect(screen).toMatchSnapshot();
  });
});
