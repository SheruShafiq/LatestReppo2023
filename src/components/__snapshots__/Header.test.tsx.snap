// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonAppBar > renders the menu button when the viewport width is less than or equal to 900 1`] = `Screen {}`;
